rickshaw-run

```diff
@@ -362,271 +626,130 @@ sub build_container_image {
-            $workshop_cmd .= " --label " . $image_md5;
-            printf "Going to generate a new client-server container image with this workshop cmd:\n\n %s\n", $workshop_cmd;
-            printf "This may take a few minutes\n";
-            ($workshop_cmd, my $workshop_output, my $workshop_rc) = run_cmd($workshop_cmd);
-            my @workshop_output = split(/\n/, $workshop_output);
-            if (open(WORKSHOP_LOG_FH, ">", $run_dir . "/workshop." . $userenv . ".out")) {
-                printf WORKSHOP_LOG_FH "%s\n", join("\n", @workshop_output);
-                close(WORKSHOP_LOG_FH);
-            } else {
-                printf "Failed to open " . $run_dir . "/workshop." . $userenv . ".out for writing\n";
-            }
-            if ($workshop_rc > 0) {
-                printf "Workshop build failed:\n";
-                printf "%s\n", join("\n", @workshop_output);
-                exit 1;
-            }
-            debug_log(sprintf "%s\n", join("\n", @workshop_output));
-            # Becasue there can be a lot of non-JSON debug/info stuff in the output before the
-            # actual JSON, find the JSON by starting at the end and scanning backwards
-            my $workshop_json = "";
-            for (my $i = scalar @workshop_output - 1; $i > 0; $i--) {
-                $workshop_json = $workshop_output[$i] . $workshop_json;
-                # Break out if we found all of the JSON
-                # In this case the JSON begins with an array '['
-                last if ($workshop_output[$i] eq "[");
-            }
-            my $coder = JSON::XS->new;
-            my $workshop_ref = $coder->decode($workshop_json);
-            $workshop_image_id = $$workshop_ref[0]{'id'};
-            printf "Workshop complete.  New image has image id of '%s'.\n", $workshop_image_id;
-            printf "Now pushing to container registry with this command:\n%s\n", $push_cmd;
-            ($push_cmd, my $output, my $buildah_rc) = run_cmd($push_cmd);
-            if ($buildah_rc > 0) {
-                printf "Pushing workshop image failed:\n%s\n", $output;
-                exit 1;
-            } else {
-                printf "Removing local container with this command:\n%s\n", $rmi_cmd;
-                ($rmi_cmd, $output, $buildah_rc) = run_cmd($rmi_cmd);
-                if ($buildah_rc > 0) {
-                    printf "Removing workshop image failed:\n%s\n", $output;
-                    exit 1;
-                }
-            }
+    }
+    $image_url = $run{'reg-repo'} . ":" . $workshop_args[$i]{'tag'};
+    # After finding the most complete image, build any "more" complete images until "most"
+    # complete is built.
+    $i++;
+    if ($i < scalar @workshop_args) {
+        while ($i < scalar @workshop_args) {
+            workshop_build_image($workshop_args[$i]{'userenv'}, $workshop_args[$i]{'reqs'}, $workshop_args[$i]{'tag'}, $workshop_args[$i]{'skip_update'});
+            push_local_image($workshop_args[$i]{'tag'});
```

Comment:
Should we be checking for errors here or are we just doing hard exits on errors and calling that good?

Reply:
I think we just have a die() in these functions we call